radix-engine/src/blueprints/epoch_manager/epoch_manager.rs

```diff
@@ -75,6 +77,19 @@ pub struct ProposalStatistic {
     pub missed: u64,
 }
 
+impl ProposalStatistic {
+    /// A ratio of successful to total proposals.
+    /// There is a special case of a validator which did not have a chance of leading even a single
+    /// round of consensus - currently we assume they should not be punished (i.e. we return `1.0`).
+    pub fn success_ratio(&self) -> Decimal {
+        let total = self.made + self.missed;
+        if total == 0 {
+            return Decimal::one();
+        }
+        Decimal::from(self.made) / Decimal::from(total)
```

Comment:
It could be, but I am not sure it is worth it...
We have to make a pretty arbitrary decision (i.e. "if someone lead 0 rounds, is his reliability 1.0 or 0.0?") and here is a really nice single place to capture it.

If this ever hurts us, then one possible refactor [that would maintain both the precision and the semantics] is to introduce and use some `Rational64` data type for transfer.